lib/Pipeline/Runner.cpp

```diff
@@ -75,9 +77,9 @@ static void explainPipeline(const ContainerToTargetsMap &Targets,
   ExplanationLogger << Requirements.back().ToExecute->getName() << ":\n";
   prettyPrintStatus(Targets, ExplanationLogger, 2);
 
-  for (size_t I = Requirements.size(); I != 1; I--) {
-    StringRef StepName = Requirements[I - 2].ToExecute->getName();
-    const auto &TargetsNeeded = Requirements[I - 1].Objectives;
+  for (size_t I = Requirements.size(); I != 0; I--) {
+    StringRef StepName = Requirements[I - 1].ToExecute->getName();
+    const auto &TargetsNeeded = Requirements[I - 1].Input;
```

Comment:
I see magic numbers changing here, which is usually scary.
Can you briefly summarize what's going on?

Reply:
we were operating on outputs of the previous step to deduce inputs of the current one. The induction variable started from end and walked backward, so I -1 is the current guy, I-2 is the guy which outputs feed into the current guy.

This has been changed to operate in the space of inputs, rather than outputs that are transformed into inputs for the next step, so there is no need any longer to inspect the I-2 guy